# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/nvidia-open/hotfix-linux-6.16.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- open-gpu-kernel-modules-590.48.01/kernel-open/common/inc/nv-timer.h.vanilla
+++ open-gpu-kernel-modules-590.48.01/kernel-open/common/inc/nv-timer.h
@@ -23,6 +23,7 @@
 #include <linux/timer.h>
 #include <linux/kernel.h> // For container_of
+#include <linux/version.h>
 
 #include "conftest.h"
 
@@ -54,7 +54,7 @@ static inline void nv_timer_delete_sync(struct timer_list *timer)
 {
-#if !defined(NV_BSD) && NV_IS_EXPORT_SYMBOL_PRESENT_timer_delete_sync
+#if !defined(NV_BSD) && (NV_IS_EXPORT_SYMBOL_PRESENT_timer_delete_sync || LINUX_VERSION_CODE >= KERNEL_VERSION(6, 15, 0))
     timer_delete_sync(timer);
 #else
     del_timer_sync(timer);
--- open-gpu-kernel-modules-590.48.01/kernel-open/nvidia/nv-caps.c.vanilla
+++ open-gpu-kernel-modules-590.48.01/kernel-open/nvidia/nv-caps.c
@@ -33,6 +33,7 @@
 /* sys_close() or __close_fd() */
 #include <linux/syscalls.h>
+#include <linux/version.h>
 
 #define NV_CAP_DRV_MINOR_COUNT 8192
@@ -619,8 +619,8 @@
-#if NV_IS_EXPORT_SYMBOL_PRESENT_close_fd
+#if NV_IS_EXPORT_SYMBOL_PRESENT_close_fd || LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)
     close_fd(fd);
-#elif NV_IS_EXPORT_SYMBOL_PRESENT___close_fd
+#elif NV_IS_EXPORT_SYMBOL_PRESENT___close_fd || LINUX_VERSION_CODE >= KERNEL_VERSION(4, 17, 0)
     __close_fd(current->files, fd);
 #else
     sys_close(fd);
--- open-gpu-kernel-modules-590.48.01/kernel-open/nvidia/nv-nano-timer.c.vanilla
+++ open-gpu-kernel-modules-590.48.01/kernel-open/nvidia/nv-nano-timer.c
@@ -146,7 +146,7 @@
 #if NV_NANO_TIMER_USE_HRTIMER
-#if NV_IS_EXPORT_SYMBOL_PRESENT_hrtimer_setup
+#if NV_IS_EXPORT_SYMBOL_PRESENT_hrtimer_setup || LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0)
     hrtimer_setup(&nv_nstimer->hr_timer, &nv_nano_timer_callback_typed_data,
                   CLOCK_MONOTONIC, HRTIMER_MODE_REL);
 #else
--- open-gpu-kernel-modules-590.48.01/kernel-open/nvidia-modeset/nvidia-modeset-linux.c.vanilla
+++ open-gpu-kernel-modules-590.48.01/kernel-open/nvidia-modeset/nvidia-modeset-linux.c
@@ -24,6 +24,7 @@
 #include <linux/module.h>
+#include <linux/version.h>
 #include <linux/kernel.h>
@@ -2195,7 +2195,7 @@
-#if !defined(NV_BSD) && NV_IS_EXPORT_SYMBOL_PRESENT_timer_delete_sync
+#if !defined(NV_BSD) && (NV_IS_EXPORT_SYMBOL_PRESENT_timer_delete_sync || LINUX_VERSION_CODE >= KERNEL_VERSION(6, 15, 0))
             if (timer_delete_sync(&timer->kernel_timer) == 1) {
